config/filter.d/zoneminder.conf

```diff
@@ -1,27 +1,42 @@
 # Fail2Ban filter for Zoneminder login failures
 
+#Use the following command for testing - Ubuntu
+#sudo fail2ban-regex -v /var/log/zm/web_php.log /etc/fail2ban/filter.d/zoneminder.conf
+
+# Example pattern ZM < 1.34
+#pattern=[Wed Apr 27 23:12:07.736196 2016] [:error] [pid 2460] [client 10.1.1.1:47296] WAR [Login denied for user "test"], referer: https://zoneminderurl/index.php
+
+# Example pattern ZM > 1.34 = Default pattern with no additional settings with in ZM Options Tab
+#pattern=7/17/22, 3:01:06 PM GMT+1.104844 web_php[281296].ERR [192.168.0.3] [Could not retrieve user test details] at /usr/share/zoneminder/www/includes/auth.php line 312
+
+# Example pattern ZM > 1.34 with ZM Options Tab setting DATETIME_FORMAT_PATTERN = yyyy/MM/dd HH:mm:ss
+#pattern=2022/08/18 18:26:02.873802 web_php[918957].ERR [192.168.0.181] [Could not retrieve user test details] at /usr/share/zoneminder/www/includes/auth.php line 312
+
+
 [INCLUDES]
 before = apache-common.conf
 
 [Definition]
 
-# patterns: [Mon Mar 28 16:50:49.522240 2016] [:error] [pid 1795] [client 10.1.1.1:50700] WAR [Login denied for user "username1"], referer: https://zoneminder/
-#           [Sun Mar 28 16:53:00.472693 2021] [php7:notice] [pid 11328] [client 10.1.1.1:39568] ERR [Could not retrieve user test details], referer: https://zm/
-#           [Sun Mar 28 16:59:14.150625 2021] [php7:notice] [pid 11336] [client 10.1.1.1:39654] ERR [Login denied for user "john"], referer: https://zm/
-#
 # Option:  failregex
-# Notes.:  regex to match the login failure and non-existent user error messages in the logfile.
 
-prefregex = ^%(_apache_error_client)s (?:ERR|WAR) <F-CONTENT>\[(?:Login denied|Could not retrieve).*</F-CONTENT>$
+#Uncomment the below regex for use with ZM < 1.34
+#failregex = ^%(_apache_error_client)s WAR \[Login denied for user "[^"]*"\]
+
+#Uncomment the below regex for use with ZM > 1.34
+failregex = ^\s*web_php\[\d+\]\.ERR \[<HOST>\] \[(?:Could not retrieve user|Login denied for user) \S+
```

Comment:
Since the regex is different, one'd need to adjust the test-cases in [fail2ban/tests/files/logs/zoneminder](https://github.com/fail2ban/fail2ban/blob/master/fail2ban/tests/files/logs/zoneminder).